src/Orleans.Runtime/Utilities/UnobservedExceptionsHandler.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Orleans.Runtime.Scheduler;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans.Runtime
+{
+    internal static class ServiceProviderExtensions
+    {
+        public static void InitializeSiloUnobservedExceptionsHandler(this IServiceProvider services)
+        {
+            //resolve handler from DI to initialize it
+            var ignore = services.GetService<SiloUnobservedExceptionsHandler>();
+        }
+    }
+
+    internal class SiloUnobservedExceptionsHandler : IDisposable
+    {
+        private readonly ILogger logger;
+        
+        public SiloUnobservedExceptionsHandler(ILogger<SiloUnobservedExceptionsHandler> logger)
+        {
+            this.logger = logger;
+            AppDomain.CurrentDomain.UnhandledException += this.DomainUnobservedExceptionHandler;
+            TaskScheduler.UnobservedTaskException += InternalUnobservedTaskExceptionHandler;
+        }
+
+        private void InternalUnobservedTaskExceptionHandler(object sender, UnobservedTaskExceptionEventArgs e)
+        {
+            var aggrException = e.Exception;
+            var baseException = aggrException.GetBaseException();
+            var tplTask = (Task)sender;
+            var contextObj = tplTask.AsyncState;
+            var context = contextObj as ISchedulingContext;
+
+            try
+            {
+                UnobservedExceptionHandler(context, baseException);
+            }
+            finally
+            {
+                if (e.Observed)
+                {
+                    logger.Info(ErrorCode.Runtime_Error_100311, "UnobservedExceptionsHandlerClass caught an UnobservedTaskException which was successfully observed and recovered from. BaseException = {0}. Exception = {1}",
```

Comment:
`UnobservedExceptionsHandlerClass` does no longer exist. Just say `Silo` anyway, no need to be explicit about this internal type (even though it will be part of the logger category anyway)